File: src/Accounting/Vns.Erp.Core.Accounting.VbAccountingGUI/CCDC/frmKhaiBaoCCDC_DanhSach.vb
Imports Vns.Erp.Core.Admin.Domain
Imports Vns.Erp.Core.Admin.Service.Interface

Public Class frmKhaiBaoCCDC_DanhSach

#Region "Properties"
    Private _CcDmCcdcService As ICcDmCcdcService
    Public Property CcDmCcdcService As ICcDmCcdcService
        Get
            Return _CcDmCcdcService
        End Get
        Set(ByVal value As ICcDmCcdcService)
            _CcDmCcdcService = value
        End Set
    End Property


    Private _CcCtCcdcService As ICcCtCcdcService
    Public Property CcCtCcdcService() As ICcCtCcdcService
        Get
            Return _CcCtCcdcService
        End Get
        Set(ByVal value As ICcCtCcdcService)
            _CcCtCcdcService = value
        End Set
    End Property


    Private _CcCtCcdcNvonService As ICcCtCcdcNvonService
    Public Property CcCtCcdcNvonService() As ICcCtCcdcNvonService
        Get
            Return _CcCtCcdcNvonService
        End Get
        Set(ByVal value As ICcCtCcdcNvonService)
            _CcCtCcdcNvonService = value
        End Set
    End Property


#End Region

#Region "Variables"
    Private CCDC_id As Guid
    Private objDmCcdc As CcDmCcdc
#End Region

#Region "Contructor"
    Public Sub New()
        InitializeComponent()
        TNCommon.setIconControl(Me)
    End Sub
#End Region

#Region "Functions"

    Private Sub DoLoadData()
        Dim lstCCDC As List(Of CcDmCcdc) = New List(Of CcDmCcdc)
        lstCCDC.AddRange(_CcDmCcdcService.GetAllByDonviID(Generals.DON_VI.Id))
        grcCCDC.DataSource = lstCCDC
    End Sub

    Private Sub grv_focus()
        TNCommon.GridView_SetRowFocus(grvCCDC, "MaCcdc", objDmCcdc.MaCcdc)
    End Sub

    Private Sub DoCreate()
        Dim frm As frmKhaiBaoCCDC_ChiTiet = CType(ObjectFactory.GetObject("frmKhaiBaoCCDC_ChiTiet"), frmKhaiBaoCCDC_ChiTiet)
        objDmCcdc = frm.Show_Form(Nothing, DataInputState.AddMode)
        If objDmCcdc IsNot Nothing Then
            Me.DoLoadData()
            grv_focus()
        End If
    End Sub

    Private Sub DoModify()
        If grvCCDC.RowCount <> 0 Then
            Dim i As Integer = grvCCDC.FocusedRowHandle

            If i < 0 Then
                Message_Warning("Không có bản ghi nào được lựa chọn")
                Return
            End If

            CCDC_id = CType(grvCCDC.GetRow(i), CcDmCcdc).Id

            Dim frm As frmKhaiBaoCCDC_ChiTiet = CType(ObjectFactory.GetObject("frmKhaiBaoCCDC_ChiTiet"), frmKhaiBaoCCDC_ChiTiet)
            objDmCcdc = frm.Show_Form(CCDC_id, DataInputState.EditMode)
            If objDmCcdc IsNot Nothing Then
                Me.DoLoadData()
                grvCCDC.FocusedRowHandle = i
            End If
        Else
            Message_Warning("Không có bản ghi nào được lựa chọn")
        End If
    End Sub

    Private Sub DoDelete()
        If grvCCDC.RowCount <> 0 Then

            Dim i As Integer = grvCCDC.FocusedRowHandle

            If i < 0 Then
                Message_Warning("Không có bản ghi nào được lựa chọn")
                Return
            End If
            CCDC_id = CType(grvCCDC.GetRow(i), CcDmCcdc).Id
            If Message_Confirm("Bạn chắc chắn muốn xóa bản ghi này?") Then
                CcDmCcdcService.DeleteCCDC(CCDC_id)
                Me.DoLoadData()
            End If
        Else
            Message_Warning("Không có bản ghi nào được lựa chọn")
        End If
    End Sub

#End Region

#Region "Events"

    Private Sub frmKhaiBaoCCDC_DanhSach_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            DoLoadData()
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub

    Private Sub btnAdd_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAdd.Click
        Try
            DoCreate()
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub

    Private Sub btnModify_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnModify.Click
        Try
            DoModify()
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub

    Private Sub btnDelete_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnDelete.Click
        Try
            DoDelete()
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub

    Private Sub btnClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub

#End Region

End Class

File: src/Asset/Vns.Erp.Core.Asset.CsAssetGUI/CCDC/frmKhaiBaoCCDC_DanhSach.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Vns.Erp.Core.Asset.Domain;
using Vns.Erp.Core.Asset.Service.Interface;

namespace Vns.Erp.Core.Asset.CsAssetGUI
{
    public partial class frmKhaiBaoCCDC_DanhSach
    {

        #region "Properties"
        private ICcDmCcdcService _CcDmCcdcService;
        public ICcDmCcdcService CcDmCcdcService
        {
            get { return _CcDmCcdcService; }
            set { _CcDmCcdcService = value; }
        }


        private ICcCtCcdcService _CcCtCcdcService;
        public ICcCtCcdcService CcCtCcdcService
        {
            get { return _CcCtCcdcService; }
            set { _CcCtCcdcService = value; }
        }


        private ICcCtCcdcNvonService _CcCtCcdcNvonService;
        public ICcCtCcdcNvonService CcCtCcdcNvonService
        {
            get { return _CcCtCcdcNvonService; }
            set { _CcCtCcdcNvonService = value; }
        }


        #endregion

        #region "Variables"
        private Guid CCDC_id;
        #endregion
        private CcDmCcdc objDmCcdc;

        #region "Contructor"
        public frmKhaiBaoCCDC_DanhSach()
        {
            InitializeComponent();

            LoadEvents();

            TNCommon.setIconControl(this);
        }

        private void LoadEvents()
        {
            this.Load += frmKhaiBaoCCDC_DanhSach_Load;
            btnAdd.Click += new EventHandler(btnAdd_Click);
            btnModify.Click += new EventHandler(btnModify_Click);
            btnDelete.Click += new EventHandler(btnDelete_Click);
            btnClose.Click += new EventHandler(btnClose_Click);
        }
        #endregion

        #region "Functions"

        private void DoLoadData()
        {
            List<CcDmCcdc> lstCCDC = new List<CcDmCcdc>();
            lstCCDC.AddRange(_CcDmCcdcService.GetAllByDonviID(Generals.DON_VI.Id));
            grcCCDC.DataSource = lstCCDC;
        }

        private void grv_focus()
        {
            TNCommon.GridView_SetRowFocus(grvCCDC, "MaCcdc", objDmCcdc.MaCcdc);
        }

        private void DoCreate()
        {
            frmKhaiBaoCCDC_ChiTiet frm = (frmKhaiBaoCCDC_ChiTiet)ObjectFactory.GetObject("frmKhaiBaoCCDC_ChiTiet");
            objDmCcdc = frm.Show_Form(Null.NullGuid, FormGlobals.DataInputState.AddMode);
            if (objDmCcdc != null)
            {
                this.DoLoadData();
                grv_focus();
            }
        }

        private void DoModify()
        {
            if (grvCCDC.RowCount != 0)
            {
                int i = grvCCDC.FocusedRowHandle;

                if (i < 0)
                {
                    FormGlobals.Message_Warning("Không có bản ghi nào được lựa chọn");
                    return;
                }

                CCDC_id = ((CcDmCcdc)grvCCDC.GetRow(i)).Id;

                frmKhaiBaoCCDC_ChiTiet frm = (frmKhaiBaoCCDC_ChiTiet)ObjectFactory.GetObject("frmKhaiBaoCCDC_ChiTiet");
                objDmCcdc = frm.Show_Form(CCDC_id, FormGlobals.DataInputState.EditMode);
                if (objDmCcdc != null)
                {
                    this.DoLoadData();
                    grvCCDC.FocusedRowHandle = i;
                }
            }
            else
            {
                FormGlobals.Message_Warning("Không có bản ghi nào được lựa chọn");
            }
        }

        private void DoDelete()
        {

            if (grvCCDC.RowCount != 0)
            {
                int i = grvCCDC.FocusedRowHandle;

                if (i < 0)
                {
                    FormGlobals.Message_Warning("Không có bản ghi nào được lựa chọn");
                    return;
                }
                CCDC_id = ((CcDmCcdc)grvCCDC.GetRow(i)).Id;
                if (FormGlobals.Message_Confirm("Bạn chắc chắn muốn xóa bản ghi này?"))
                {
                    CcDmCcdcService.DeleteCCDC(CCDC_id);
                    this.DoLoadData();
                }
            }
            else
            {
                FormGlobals.Message_Warning("Không có bản ghi nào được lựa chọn");
            }
        }

        #endregion

        #region "Events"

        private void frmKhaiBaoCCDC_DanhSach_Load(System.Object sender, System.EventArgs e)
        {
            try
            {
                DoLoadData();
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }

        private void btnAdd_Click(System.Object sender, System.EventArgs e)
        {
            try
            {
                DoCreate();
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }

        private void btnModify_Click(System.Object sender, System.EventArgs e)
        {
            try
            {
                DoModify();
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }

        private void btnDelete_Click(System.Object sender, System.EventArgs e)
        {
            try
            {
                DoDelete();
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }

        private void btnClose_Click(System.Object sender, System.EventArgs e)
        {
            this.Close();
        }

        #endregion

    }

}
